### ISSUE TITLE
Incorrect model rating

### ISSUE THREAD

**Issue by xiamu333** [SebOchs/SAF#2]

**Comment by SebOchs** [comment]
Hi,
can you provide more details for your issue?
There is no student answer "0" in the dataset for that specific question.

**Comment by xiamu333** [comment]
Thank you for your reply!
In "test. json", I tried various student answers, even numbers 0 or spaces. But the model has been consistently rated 1 point and feedback is “korrekt!”.
Have you ever experienced this situation before? Where should the problem lie?
Thanks again!

**Comment by SebOchs** [comment]
I 'm a bit confused by your questions, there is no test.json file for this dataset.
Did you convert it to one with the datasets library from huggingface?
If yes, it seems the student answers were converted incorrectly, given your example.
Have you fine-tuned a model on the dataset and it makes incorrect guesses on the test set?
Then it may be because the student answer should not be "0", but a short text, then you have to look into your conversion method.
If that is not the case, the performance of your utilized model may be lacking if it always predicts 1.

**Comment by xiamu333** [comment]
Sincerely thank you for your reply.

In the inference stage, I entered a "test.json" required for the inference code. After the inference was completed, the model output a "result.json", but the rating and feedback were very unreasonable. Do you know roughly where the problem occurred? The "result. json" is shown as follows:

{"Question":
	{"0":"Worauf sollst du in Bezug auf das Gesamterscheinungsbild des Shops achten\uff1f",
	"1":"Worauf sollst du in Bezug auf das Gesamterscheinungsbild des Shops achten\uff1f",
	"2":"Worauf sollst du in Bezug auf das Gesamterscheinungsbild des Shops achten\uff1f"},
"Reference Answer":
	{"0":"Das Gesamterscheinungsbild besteht aus der Sauberkeit des Shops und der Ausstattung mit Waren.Bei der Sauberkeit des Shops ist beispielsweise auf die Sauberkeit der Fu\\u00dfb\\u00f6den oder der Br\\u00f6tchenauslage zu achten.Der bautechnische Zustand des Shops soll hier nicht ber\\u00fccksichtigt werden\uff01Bei der Ausstattung 	mit Waren soll darauf geachtet werden\uff0c ob die Regale vollst\\u00e4ndig mit Produkten best\\u00fcckt sind und ob der Kunde eine Wertigkeit oder einen Nutzen in den Produkten sieht.",
	"1":"Das Gesamterscheinungsbild besteht aus der Sauberkeit des Shops und der Ausstattung mit Waren.Bei der Sauberkeit des Shops ist beispielsweise auf die Sauberkeit der Fu\\u00dfb\\u00f6den oder der Br\\u00f6tchenauslage zu achten.Der bautechnische Zustand des Shops soll hier nicht ber\\u00fccksichtigt werden\uff01Bei der Ausstattung 	mit Waren soll darauf geachtet werden\uff0c ob die Regale vollst\\u00e4ndig mit Produkten best\\u00fcckt sind und ob der Kunde eine Wertigkeit oder einen Nutzen in den Produkten sieht.",
	"2":"Das Gesamterscheinungsbild besteht aus der Sauberkeit des Shops und der Ausstattung mit Waren.Bei der Sauberkeit des Shops ist beispielsweise auf die Sauberkeit der Fu\\u00dfb\\u00f6den oder der Br\\u00f6tchenauslage zu achten.Der bautechnische Zustand des Shops soll hier nicht ber\\u00fccksichtigt werden\uff01Bei der Ausstattung 	mit Waren soll darauf geachtet werden\uff0c ob die Regale vollst\\u00e4ndig mit Produkten best\\u00fcckt sind und ob der Kunde eine Wertigkeit oder einen Nutzen in den Produkten sieht."},
"Student Answer":
	{"0":"Keine Ahnung.",
	"1":"Beim Gesamterscheinungsbild des Shops ist auf eine klare Raumaufteilung, ansprechende Beleuchtung, markenbezogene Einheitlichkeit (z. B. Farben, Logos) und eine \u00fcbersichtliche Pr\u00e4sentation der Waren zu achten.",
	"2":"Das Gesamterscheinungsbild besteht aus der Sauberkeit des Shops und der Ausstattung mit Waren.Bei der Sauberkeit des Shops ist beispielsweise auf die Sauberkeit der Fu\\\\u00dfb\\\\u00f6den oder der Br\\\\u00f6tchenauslage zu achten.Der bautechnische Zustand des Shops soll hier nicht ber\\\\u00fccksichtigt werden\uff01Bei der Ausstattung mit Waren soll darauf geachtet werden\uff0c ob die Regale vollst\\\\u00e4ndig mit Produkten best\\\\u00fcckt sind und ob der Kunde eine Wertigkeit oder einen Nutzen in den Produkten sieht."},
"Score":
	{"0":"1.0",
	"1":"1.0",
	"2":"1.0"},
"Feedback":
	{"0":" korrekt!",
	"1":" korrekt!",
	"2":" korrekt!"}}

**Comment by SebOchs** [comment]
Ah my bad. I haven't looked into my own code for a while.
Did you fine-tune the model with the code? 
This is expected behaviour for untrained or low-quality models.

**Comment by xiamu333** [comment]
My steps were all carried out according to your guidance in the "README". The first step is preprocessing. After the fine-tuning of the model in the second step is completed, a file will be generated called "models/wq_stcore/wq_stcore_mT5_ger_epoch=18-my_stetric=0.2091. ckpt". The third step is testing, and the fourth step is bert_storing. After the inference in the fifth step, the model will output a file called "result. json" based on the file "test. json" I input. The final output is the unreasonable result I sent you before.

Thank you for handling my issue.

![Image](https://github.com/user-attachments/assets/b25ce10c-e0cc-4367-a502-aa668bc2dd0d)

**Comment by SebOchs** [comment]
What metric results did you get when running testing.py? 
If the results were not good, this would explain the model outputs.
Also, please note that our baselines were not meant for state of the art performances on the dataset/task, but rather a first impression of what was achievable at the time (> 3 years ago) with our compute.  
The dataset is quite challenging and newer models / a better training setup might achieve better results than our baseline setup.
